### ISSUE TITLE
After ~8h of operation, unexpected shutdown, without any warning or error message, twice

### ISSUE THREAD

**Issue by pwrml** [PurpleI2P/i2pd#830]
Hi!

I installed latest github repo i2pd this past Monday, and started i2pd. It shut down, without any warning or error message, after about 8 hours.

I noticed Wednesday, and started it again, and it did the same. On the first run it was not a floodfill, and on the second run it was.

Is this a known error?

Can you please give me some pointers to how to track down the error?

Someone on #i2p on FreeNode said they had had a similar problem, that it shuts down after a week they said, others had had no problems.

It's important that i2pd works, I hope we can weed out this one soon.

I'll run i2pd again, this time as "gdb --args i2pd --floodfill --loglevel=debug .

**Comment by r4sas** [comment]
Please install i2pd-dbg package from repo, and add `ulimit -c unlimited` in `/etc/default/i2pd` and restart i2pd.
When i2pd will crash, gdb will create dump in `/var/lib/i2pd/core`.
To get information from dump, go to `/var/lib/i2pd` and run `gdb i2pd core`.
We need output of commands `where` and `info threads`.

**Comment by LLE8** [comment]
Dump file may be large, up to 300MB or even more

**Comment by majestrate** [comment]
OS/Arch?

**Comment by orignal** [comment]
>I installed latest github repo i2pd this past Monday
Trunk or 2.12.0 ?

**Comment by LLE8** [comment]
@pwrml 
I have a similar problem
https://github.com/PurpleI2P/i2pd/issues/799

**Comment by orignal** [comment]
@LLE8 try to build trunk

**Comment by pwrml** [comment]
@majestrate , OpenBSD 6.0.

@LLE8 , I think it just quit, no core file. Which is consistent with my understanding below that it was quit by SIGPIPE (and not a code error, which would have lead to SIGSEGV and a core file).

@r4sas , let's do it from the console.

I tried "gdb --args i2pd --floodfill --loglevel=debug" as I said above.

I just checked back, and the i2pd had quit to gdb with reference to a SIGPIPE signal.

I just did "cont" in gdb as SIGPIPE not is a real error by any standards, and i2pd looks like it's doing well again now. 

SIGPIPE is discussed here: http://www.tedunangst.com/flak/post/SIGPIPE-can-happen-to-you

A first look in i2pd's sourcecode shows me that i2pd not installs any SIGPIPE handler.

Bug??

I just started the separate issue https://github.com/PurpleI2P/i2pd/issues/833 to discuss this. Hopefully that is the only problem i2pd has. Looking forward to have I2P run perfectly stably!
